pkg/resource/deploy/step_generator.go

```diff
@@ -38,6 +39,78 @@ import (
 	"github.com/pulumi/pulumi/sdk/v3/go/common/util/result"
 )
 
+// ContinueResourceDiffEvent is a step that asks the engine to continue provisioning a resource after completing its
+// diff, it is always created from a base RegisterResourceEvent.
+type ContinueResourceDiffEvent interface {
```

Comment:
Guessing there's nothing we have to think about in terms of these events getting to snapshots/journaling code etc.? I remember some bits of the codebase elsewhere about "ephemeral" steps but suspect that is a separate set of types -- figured I'd double check though.